### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by uschi2000**
@markelliot @jared2501

### CODE REVIEW

**Comment by jared2501** [comment]
is it worth making this a multi project gradle project?

File: junit-resource-poller/src/main/java/com/palantir/junit/PollableResource.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+import com.google.common.base.Optional;
+
+public interface PollableResource {
+    /**
+     * Returns {@link Optional#absent} iff the the resource is available, or an exception explaining why the resource is
+     * not considered ready else.
+     */
+    Optional<Exception> isReady();
```
Comment (markelliot) [inline junit-resource-poller/src/main/java/com/palantir/junit/PollableResource.java:27]: there's really nothing wrong with this, but it seems strange that the positive result of this method is an `Optional.absent()`

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.junit;
+
+import com.google.common.base.Optional;
+
+
+public abstract class ResourcePoller {
```
Comment (markelliot) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: final w/ private constructor?

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Returns when the given target resource is {@link PollableResource#isReady()} after {@code timeoutMillis}
+     * milliseconds, or throws the last error returned from polling otherwise.
+     */
+    public static void poll(long timeoutMillis, long intervalMillis, PollableResource target) throws Exception {
```
Comment (markelliot) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: ...or maybe mark this as final?

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        long startTime = System.currentTimeMillis();
+        Optional<Exception> lastException;
+        do {
+            lastException = target.isReady();
+            Thread.sleep(intervalMillis);
```
Comment (markelliot) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: I suspect we'd be better off using a `ScheduledExecutor`

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java (hunk 2 above)
Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: done

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java (hunk 3 above)
Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: done

File: junit-resource-poller/src/main/java/com/palantir/junit/PollableResource.java (hunk 1 above)
Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/PollableResource.java:27]: yeah, odd, but I don't really know a better naming scheme here.

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java (hunk 4 above)
Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: done

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+public final class ResourcePoller {
+
+    private ResourcePoller() {}
+
+    /**
+     * Calls {@link PollableResource#isReady()} at most {@code numAttempts} times and returns once the given target
+     * resource returns {@link Optional#absent()} to indicate that it is ready. Throws the last exception returned by
+     * {@link PollableResource#isReady()} otherwise.
+     */
+    public static void poll(int numAttemts, long intervalMillis, final PollableResource target) throws Exception {
```
Comment (markelliot) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java:35]: "attemts" -> "attempts"

Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java:35]: done

**Pull request merged**